### PR TITLE
V1 dplyr fix

### PR DESCRIPTION
**Pull request by elinw**
This should fix  dplyr filter() that is changed by the version 0.8.0 release of dplyr which is scheduled for February 1, 2019.    See https://github.com/tidyverse/dplyr/issues/4051   for an explanation.  In the new version of filter the custom class skim_df is not stripped from the returned object which leads R to attempt to use print.skim_df instead of print.data.frame. This fix removes the skim_df class from the returned object.

### CODE REVIEW

**Comment by hadley** [comment]
Why should filtering strip the class? That seems like a surprising action to me.

**Comment by elinw** [comment]
@hadley it seemed surprising to me too!  I'm not sure what change in filter (and maybe mutate, haven't tested yet) caused this.

**Comment by hadley** [comment]
You are now deliberately stripping the class — I don't understand why that is necessary.

**Comment by elinw** [comment]
Because if the class is there, R will use the print method for that class rather than the data frame method, hence with filter, specifically in the example in our vignette, the print errors because it can't run the print method which assumes that specific things will be present which are not.  The skim object is a long object with 6 columns (variable, type, stat, level, value, formatted)  and then for a given variable being skimmed has a row for each statistic.  When we filter on e.g. `stat == "mean"`   all those statistics are no longer present.  Now it should be the case that the skim print method actually looks at what rows are present before trying to print them, but it  doesn't.    None of this mattered when the class was getting stripped because the print was   using the data frame print.  

So basically, I think the new behavior is probably what should have been the behavior all along, we just unknowingly were taking advantage of the fact that this stripping was happening.   So you essentially fixed a bug (I could definitely see people complaining about the stripping) and that broke our behavior.  I actually think it would probably be helpful to our users long term to have the filtered data sets still marked as skim_df if we weren't abandoning that whole approach for v2.

**Comment by hadley** [comment]
Ok, makes sense. In the future, you might look at what googledrive does, as it has a similar problem (i.e. a tibble that has "magical" columns that must exist or it doesn't behave like a dribble). Hopefully, a future version of tibble will provide more support for this fairly common use case.
